### ISSUE TITLE
Federated metadata is not loading in metadata-providers.xml

### ISSUE THREAD

**Issue by mzico** [GluuFederation/oxShibboleth#36]
Current metadata-providers.xml.vm can't load any federated metadata. 
A workaround is removing `generate` method and write new 'metadata-providers.xml.vml` like this:  https://github.com/GluuFederation/support-docs/blob/master/troubleshooting/shibboleth/metadata_not_loading.md

We need to re-write 'generate' method properly as federated metadata can be loaded. It's happening in 3.1.1 as well.

**Comment by mzico** [comment]
@dmogn : this has to be fixed asap. I see this issue still in 3.1.2.

**Comment by aliaksander-samuseu** [comment]
Hi, @mzico & @dmogn 

Here is how I fixed it in 3.1.1 in my instance. 3.1.2 seems to use exactly the same template, so it should work for it as well. Note it's just my ad-hoc fix and it wasn't thoroughly tested, may be @dmogn will find a better solution, so it's just a suggestion:
```
# diff -c /opt/gluu/jetty/identity/conf/shibboleth3/idp/metadata-providers.xml.vm /opt/gluu/jetty/identity/conf/shibboleth3/idp/metadata-providers.xml.vm.origin 
*** /opt/gluu/jetty/identity/conf/shibboleth3/idp/metadata-providers.xml.vm 2018-02-24 19:15:04.545129013 -0500
--- /opt/gluu/jetty/identity/conf/shibboleth3/idp/metadata-providers.xml.vm.origin  2018-02-26 14:02:34.857027317 -0500
***************
*** 27,36 ****
  #foreach( $trustRelationship in $trustParams.trusts )

          #set ($profileConfigMap = $trustRelationship.profileConfigurations)
! 
!         #set ($isFederation = ($trustParams.trustEntityIds.get($trustRelationship.inum).size() > 1))
! 
!         #if(!$profileConfigMap.isEmpty() || $isFederation)

              #* #set($providerId = $StringHelper.removePunctuation($trustRelationship.inum)) *#
              #* #set($providerUrl = $trustParams.trustEntityIds.get($trustRelationship.inum).get(0)) *#
--- 27,33 ----
  #foreach( $trustRelationship in $trustParams.trusts )

          #set ($profileConfigMap = $trustRelationship.profileConfigurations)
!         #if(!$profileConfigMap.isEmpty())

              #* #set($providerId = $StringHelper.removePunctuation($trustRelationship.inum)) *#
              #* #set($providerUrl = $trustParams.trustEntityIds.get($trustRelationship.inum).get(0)) *#
```
So you need to add one new line, and edit another one.

**Comment by dmogn** [comment]
Done.
